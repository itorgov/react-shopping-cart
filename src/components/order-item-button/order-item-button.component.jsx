import React from 'react';
import classNames from 'classnames';

import {ReactComponent as IconTrash} from './trash.svg';
import {ReactComponent as IconPlus} from "./plus.svg";
import {ReactComponent as IconMinus} from "./minus.svg";
import styles from './order-item-button.module.scss';

const renderIcon = variant => {
    switch (variant) {
        case 'remove':
            return <IconTrash/>;
        case 'increment':
            return <IconPlus/>;
        case 'decrement':
            return <IconMinus/>;
        default:
            return '';
    }
};

export const OrderItemButton = ({variant, shake = false}) => (
    <button className={classNames({
        [styles.button]: true,
        [styles[variant]]: true,
        [styles.shake]: shake,
    })}
    >
        {renderIcon(variant)}
    </button>
);
