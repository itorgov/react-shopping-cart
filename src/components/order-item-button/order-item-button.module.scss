.button {
  background-color: transparent;
  border: 0;

  display: flex;
  justify-content: center;
  align-items: center;

  &:focus {
    outline: none;
  }

  & svg {
    width: 2rem;
    height: 2rem;
    fill: currentColor;
    opacity: 0.8;
    transition: all 0.2s;
  }

  &:hover svg,
  &:active svg,
  &:focus svg {
    opacity: 1;
  }
}

.remove {
  color: var(--color-secondary);
}

.increment,
.decrement {
  color: var(--color-text-dark);

  & svg {
    opacity: 0.6;
  }
}

.shake {
  &:hover,
  &:focus {
    animation: shake 1s;
  }
}

@keyframes shake {
  0% {
    transform: rotate(0);
  }

  25% {
    transform: rotate(10deg);
  }

  50% {
    transform: rotate(-10deg);
  }

  75% {
    transform: rotate(5deg);
  }

  100% {
    transform: rotate(0);
  }
}
