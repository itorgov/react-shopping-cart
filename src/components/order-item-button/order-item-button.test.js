import React from "react";
import {shallow} from "enzyme";

import {OrderItemButton} from './order-item-button.component';

describe('<OrderItemButton />', () => {
    it('expect to render', () => {
        expect(shallow(<OrderItemButton />)).toMatchSnapshot();
    });

    it('remove variant adds "remove" class', () => {
        expect(shallow(<OrderItemButton variant='remove' />).hasClass('remove')).toBe(true);
    });

    it('increment variant adds "increment" class', () => {
        expect(shallow(<OrderItemButton variant='increment' />).hasClass('increment')).toBe(true);
    });

    it('increment variant adds "decrement" class', () => {
        expect(shallow(<OrderItemButton variant='decrement' />).hasClass('decrement')).toBe(true);
    });

    it('shake variant adds "shake" class', () => {
        expect(shallow(<OrderItemButton shake={true} />).hasClass('shake')).toBe(true);
    });
});
