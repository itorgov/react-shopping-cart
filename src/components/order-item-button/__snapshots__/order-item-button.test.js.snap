// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderItemButton /> expect to render 1`] = `
<button
  className="button"
/>
`;
